lib/Bio/EnsEMBL/DataCheck/Checks/CheckMethodLinkSpeciesSetNames.pm

```diff
@@ -0,0 +1,75 @@
+=head1 LICENSE
+
+Copyright [2018-2019] EMBL-European Bioinformatics Institute
+
+Licensed under the Apache License, Version 2.0 (the 'License');
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an 'AS IS' BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+
+=cut
+
+package Bio::EnsEMBL::DataCheck::Checks::CheckMethodLinkSpeciesSetNames;
+
+use warnings;
+use strict;
+
+use Moose;
+use Test::More;
+use Bio::EnsEMBL::DataCheck::Test::DataCheck;
+
+extends 'Bio::EnsEMBL::DataCheck::DbCheck';
+
+use constant {
+  NAME           => 'CheckMethodLinkSpeciesSetNames',
+  DESCRIPTION    => 'Check for consistency of names in method_link_species_set (and species_set_header)',
+  GROUPS         => ['compara'],
+  DATACHECK_TYPE => 'critical',
+  DB_TYPES       => ['compara'],
+  TABLES         => ['genome_db', 'method_link_species_set', 'species_set', 'species_set_header']
+};
+
+sub tests {
+  my ($self) = @_;
+  my $dbc = $self->dba->dbc;
+  my $mlss_adap = $self->dba->get_MethodLinkSpeciesSetAdaptor;
+  my $mlss = $mlss_adap->fetch_all;
+
+  foreach my $mlss ( @$mlss ) {
+    my $mlss_name = $mlss->name;
+    my $mlss_id = $mlss->dbID;
+    my $species_set = $mlss->species_set;
+    my $species_set_name = $species_set->name;
+    my $species_set_id = $species_set->dbID;
+    my $gdbs = $species_set->genome_dbs;
+    my $gdb_count = scalar( @$gdbs );
+
+    if ( $mlss_name =~ /^([a-zA-Z]+) /  ) {
+      my $mlss_p1 = $1;
+      my $desc_1 = "The current convention is in place for mlss $mlss_name ($mlss_id) and species_set $species_set_name ($species_set_id)";
+      unlike( $mlss_name, qr/^(protein|nc|species)/, $desc_1 );
+      my $desc_2 = "species_set $species_set_id for mlss $mlss_name ($mlss_id) starts with the species_set name $species_set_name";
+
+      if ( $species_set_name =~ /collection/ ) {
+        is( $species_set_name, "collection-$mlss_p1", $desc_2 );
+      }
+      else {
+        is( $species_set_name, $mlss_p1, $desc_2 );
+      }
+    }
+    if ( $gdb_count >= 1 && $gdb_count <=2 ) {
+      my @species_count = split /-/, $species_set_name;
```

Comment:
Does it actually reach this ? The regexp line 54 doesn't allow hyphens. Also, some older mlss have a dot (e.g. `H.sap` instead of `Hsap`)

Reply:
It does. The `if`  statement comes after the first one. The tests report when the DC has been run also:
`ok 3364 - For Etef-Sita orthologues (302683) the species_set Etef-Sita (603156) is appropriately named with the correct number of genomes
        ok 3365 - For B.vul-T.pra orthologues (200256) the species_set B.vul-T.pra (600271) is appropriately named with the correct number of genomes`